content/en/docs/Tutorials/basic.md

```diff
@@ -0,0 +1,242 @@
+
+    "github.com/AdRoll/baker"
+)
+
+func main() {
+    components := baker.Components{/* define components */}
+    if err := baker.MainCLI(components); err != nil {
+		log.Fatal(err)
+	}
+}
+```
+
+### Define baker.Components
+
+Our `baker.Components` implementation must:
+
+* include the components we need to use
+* define `FieldByName` and `FieldName` to map our dataset
+
+The simplest way to add the components to Baker is just to add all available components:
+
+```go
+components := baker.Components{
+    Inputs:      input.All,
+    Filters:     filter.All,
+    Outputs:     output.All,
+    Uploads:     upload.All,
+    /* ... */
+}
+```
+
+As for the mapping functions, we need to map from name to `FieldIndex` and vice-versa:
+
+```go
+var fields = []string{
+	"review_num",
+	"brand",
+	"variety",
+	"style",
+	"country",
+	"stars",
+	"top_ten",
+}
+
+var fieldsByName = map[string]baker.FieldIndex{
+	"review_num": 0,
+	"brand":      1,
+	"variety":    2,
+	"style":      3,
+	"country":    4,
+	"stars":      5,
+	"top_ten":    6,
+}
+
+func fieldByName(name string) (idx baker.FieldIndex, ok bool) {
+	idx, ok = fieldsByName[name]
+	return idx, ok
+}
+
+func fieldName(idx baker.FieldIndex) string {
+	return fields[idx]
+}
+
+components := baker.Components{
+    /* ... */
+    FieldByName: fieldByName,
+    FieldName:   fieldName,
+}
+```
+
+The complete program is available in the
+[`tutorials/` folder](https://github.com/AdRoll/baker/blob/main/tutorials/basic/main.go) in
+the Baker repository.
+
+## Run the program
+
+Once the code and the configuration file are ready, we can run the topology:
+
+```sh
+$ go build -o myProgram ./main.go 
+# Test it works as expected
+$ ./myProgram -help
+# run the topology
+$ ./myProgram topology.toml
+```
+
+Among the messages that Baker prints on stdout, the stats messages are particularly interesting:
+
+```sh
+Stats: 1s[w:0 r:0] total[w:41 r:2584 u:11] speed[w:20 r:1292] errors[p:0 i:0 f:2543 o:0 u:0]
+```
+
+Take a look at the [dedicated page](/docs/how-tos/read_stats/) to learn how to read the values.
+
+## Verify the result
+
+The resulting files are splitted into multiple folders, one for each country, and then uploaded.
+
+The [`S3`](/docs/components/uploads/s3/) upload removes the files once uploaded, so you'll find
+only the folders skeleton in the output destination folder:
+
+```sh
+~ ls --tree -l /tmp/out
+drwxrwxr-x   - username 16 Nov 11:43 /tmp/out
+drwxrwxr-x   - username 16 Nov 11:43 ├── China
+drwxrwxr-x   - username 16 Nov 11:43 ├── Hong Kong
+drwxrwxr-x   - username 16 Nov 11:43 ├── Indonesia
+drwxrwxr-x   - username 16 Nov 11:43 ├── Japan
+drwxrwxr-x   - username 16 Nov 11:43 ├── Malaysia
+drwxrwxr-x   - username 16 Nov 11:43 ├── Myanmar
+drwxrwxr-x   - username 16 Nov 11:43 ├── Singapore
+drwxrwxr-x   - username 16 Nov 11:43 ├── South Korea
+drwxrwxr-x   - username 16 Nov 11:43 ├── Taiwan
+drwxrwxr-x   - username 16 Nov 11:43 ├── Thailand
+drwxrwxr-x   - username 16 Nov 11:43 └── USA
```

Comment:
nit: can't we just use `ls -l /tmp/out`

Reply:
yes we could, I just wanted to point out that all folders were empty. but they can trust us anyway :smile: